minio/minio
# Helm s3 gateway support

## Description

* These changes have been migrated from the previous chart: https://github.com/helm/charts/tree/master/stable/minio
* Added s3 support for gateway mode.
* Added new s3 block under the gateway key to house the s3 specific variable serviceEndpoint.

## Motivation and Context

* The current chart does not enable gateway mode to be set as s3. We need the ability to deploy minio as a s3 gateway via helm.

## How to test this PR?

**create IAM principal with at least the privileges listed on the page https://docs.min.io/docs/minio-gateway-for-s3.html**

`cd helm/minio/`

**create new override values file called gateway-s3.yaml containing:**
```
mode: gateway 

rootUser: "AWS_ACCESS_KEY"
rootPassword: "AWS_SECRET_KEY"

gateway:
  type: "s3"

persistence:
  enabled: false
```

**debug install as dry-run to check entrypoint is configured correctly**

`helm install miniogateways3 ./ -f ./gateway-s3.yaml --dry-run`

**test chart install**

`helm install miniogateways3 ./ -f ./gateway-s3.yaml`

follow instructions in terminal to connect to s3 using mc client. 

## Types of changes
- [ /] New feature (non-breaking change which adds functionality)

## Checklist:
- [ /] Documentation updated
